index.ts

```diff
@@ -77,22 +96,81 @@ export const sandbox = {
     const client = await _arcFunctions.tables()
     if (seedFile) await seedDb(seedFile, dynamodbClient)
 
-    await Promise.all(
-      // @ts-expect-error table has any type
-      (inv['tables-streams'] ?? []).map(({ table }) =>
-        dynamodbClient.send(
-          new UpdateTableCommand({
-            TableName: client.name(table),
-            StreamSpecification: {
-              StreamEnabled: true,
-              StreamViewType: 'NEW_AND_OLD_IMAGES',
-            },
-          })
+    const tableStreams = inv['tables-streams']
+
+    if (tableStreams?.length) {
+      const ddbStreamsClient = new DynamoDBStreamsClient({
+        region: inv.aws.region,
+        endpoint: local.url,
+        credentials,
+      })
+      // Init table streams for those defined
+      await Promise.all(
+        // @ts-expect-error table has any type
+        tableStreams.map(({ table }) =>
+          dynamodbClient.send(
+            new UpdateTableCommand({
+              TableName: client.name(table),
+              StreamSpecification: {
+                StreamEnabled: true,
+                StreamViewType: 'NEW_AND_OLD_IMAGES',
+              },
+            })
+          )
         )
       )
-    )
+      // Reset Stream defaults
+      for (const arcStream of tableStreams) {
+        shardMap[arcStream.table] = []
+        await resetTableStreams(
+          dynamodbClient,
+          ddbStreamsClient,
+          arcStream.table
+        )
+      }
+
+      await streamLoop(ddbStreamsClient)
+    }
+
+    async function streamLoop(ddbStreamsClient: DynamoDBStreamsClient) {
+      for (const key of Object.keys(shardMap)) {
+        if (shardMap[key].length) {
+          const shardItem = shardMap[key].pop()
+          if (!shardItem) continue
+          try {
+            const event = await ddbStreamsClient.send(
+              new GetRecordsCommand({
+                ShardIterator: shardItem.ShardIterator,
+              })
+            )
+            if (event.Records?.length) {
+              invoke({
+                pragma: 'tables-streams',
+                name: key,
+                payload: event,
+              })
+            }
```

Comment:
This assumes that there is at most one Lambda per table, but I think you can have any nonnegative number of tables per Lambda.

Reply:
Handled in latest commit